datalake/src/main/java/com/sequenceiq/datalake/service/sdx/StackRequestManifester.java

```diff
@@ -0,0 +1,209 @@
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StackRequestManifester.class);
+
+    private String dummySshKey = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0Rfl2G2vDs6yc19RxCqReunFgpYj+ucyLobpTCBtfDwzIbJot2Fmife6M42mBtiTmAK6x8kc"
+            + "UEeab6CB4MUzsqF7vGTFUjwWirG/XU5pYXFUBhi8xzey+KS9KVrQ+UuKJh/AN9iSQeMV+rgT1yF5+etVH+bK1/37QCKp3+mCqjFzPyQOrvkGZv4sYyRwX7BKBLleQmIVWpofpj"
+            + "T7BfcCxH877RzC5YMIi65aBc82Dl6tH6OEiP7mzByU52yvH6JFuwZ/9fWj1vXCWJzxx2w0F1OU8Zwg8gNNzL+SVb9+xfBE7xBHMpYFg72hBWPh862Ce36F4NZd3MpWMSjMmpDPh"
+            + "centos";
+
+    @Inject
+    private CloudbreakUserCrnClient cloudbreakClient;
+
+    @Value("${sdx.cluster.definition}")
+    private String clusterDefinition;
+
+    public void configureStackForSdxCluster(StackV4Request internalRequest, SdxClusterRequest sdxClusterRequest,
+            SdxCluster sdxCluster, DetailedEnvironmentResponse environment) {
+        if (internalRequest != null) {
+            addStackV4RequestAsString(sdxCluster, internalRequest);
+        } else {
+            //TODO: get detailed env service
+            StackV4Request generatedStackV4Request = setupStackRequestForCloudbreak(sdxCluster, sdxClusterRequest, environment);
+            addStackV4RequestAsString(sdxCluster, generatedStackV4Request);
+        }
+    }
+
+    public void addStackV4RequestAsString(SdxCluster sdxCluster, StackV4Request internalRequest) {
+        try {
+            LOGGER.info("Forming request from Internal Request");
+            sdxCluster.setStackRequestToCloudbreak(JsonUtil.writeValueAsString(internalRequest));
+        } catch (JsonProcessingException e) {
+            LOGGER.info("Can not parse stack request");
+            throw new BadRequestException("Can not parse stack request", e);
+        }
+    }
+
+    public StackV4Request setupStackRequestForCloudbreak(SdxCluster sdxCluster, SdxClusterRequest clusterRequest, DetailedEnvironmentResponse environment) {
+
+        String clusterTemplateJson;
+        if (sdxCluster.getStackRequest() == null) {
+            clusterTemplateJson = FileReaderUtils.readFileFromClasspathQuietly("sdx/" + "cluster-template.json");
+        } else {
+            clusterTemplateJson = sdxCluster.getStackRequest();
+        }
+        try {
+            StackV4Request stackRequest = JsonUtil.readValue(clusterTemplateJson, StackV4Request.class);
+            stackRequest.setName(sdxCluster.getClusterName());
+            TagsV4Request tags = new TagsV4Request();
+            try {
+                tags.setUserDefined(sdxCluster.getTags().get(HashMap.class));
+            } catch (IOException e) {
+                throw new BadRequestException("can not convert from json to tags");
+            }
+            stackRequest.setTags(tags);
+            stackRequest.setEnvironmentCrn(sdxCluster.getEnvCrn());
+
+            if (!CloudPlatform.YARN.name().equals(environment.getCloudPlatform()) && !environment.getNetwork().getSubnetMetas().isEmpty()) {
```

Comment:
!environment.getNetwork() -> there was a fix, it should look like this:

            if (!CloudPlatform.YARN.name().equals(environment.getCloudPlatform())	
                    && environment.getNetwork() != null	
                    && environment.getNetwork().getSubnetMetas() != null	
                    && !environment.getNetwork().getSubnetMetas().isEmpty()) {